src/HttpClientBuilder.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+namespace Amp\Http\Client;
+
+use Amp\Http\Client\Connection\ConnectionPool;
+use Amp\Http\Client\Interceptor\FollowRedirects;
+use Amp\Http\Client\Interceptor\RetryRequests;
+
+final class HttpClientBuilder
+{
+    public static function of(HttpClient $base): self
+    {
+        return new self($base);
```

Comment:
I'm concerned this may not work as expected for users. Consider the following:

```php
$client = HttpClientBuilder::buildDefault();
// $client used in DI
$revisedClient = HttpClientBuilder::of($client)
    ->followRedirects(0)
    ->retry(2);
// $revisedClient still will follow redirects and will retry 4 times.

Reply:
What's your suggestion for an improvement here?

Reply:
With the current structure, only to not allow providing a base `HttpClient` instance to the builder.

Reply:
I've removed the possibility to pass a custom base, so this is no longer a problem.